tools/autograd/gen_variable_type.py

```diff
@@ -828,6 +885,66 @@ def emit_increment_version(f: NativeFunction) -> List[str]:
             return []
         return [f'increment_version({r});' for r in cpp.return_names(f)]
 
+    def emit_fw_derivatives() -> List[str]:
+        content: List[str] = []
+        for derivative in fw_derivatives:
+            res = derivative.var_name
+
+            if_stmt = " || ".join([FW_DERIVATIVE_CHECK_TEMPLATE.substitute(req_inp=inp.name)
```

Comment:
nit: `if_stmt` isn't very descriptive; maybe `requires_fw_grad` is more descriptive?